independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java

```diff
@@ -610,17 +613,16 @@ private static Integer getAlternativePriority(InjectableBean<?> bean) {
     }
 
     List<InjectableBean<?>> getMatchingBeans(Resolvable resolvable) {
-        List<InjectableBean<?>> matching = new LinkedList<>();
+        List<InjectableBean<?>> matching = new ArrayList<>();
```

Comment:
Again, what's wrong with the `LinkedList` here? The list of matching beans is expected to be very small in 95% of cases. Moreover, the result of this method is cached for majority of usages (incl. programmatic lookup), only `BeanManager.getBeans(Type, Annotation...)` does not leverage the cache.

Reply:
Sure the benefit is really small here, but there's a benefit. More importantly, nobody else is using LinkedList so this code is forcing us to a load a class which is otherwise dead.

Besides, the ArrayList does have better iteration performance and also consumes less memory.

I would certainly not have bothered you with this change alone - not worth it. But in the bigger picture it makes sense to consistently use the better lists.

Reply:
> More importantly, nobody else is using LinkedList so this code is forcing us to a load a class which is otherwise dead.

Are you sure? I found many references in many projects and libs used in quarkus (jackson, jaeger, graphql, neo4j, vertx, etc.). That said I'm not a big fan of this data structure but it obviously has some pros and cons.

> Besides, the ArrayList does have better iteration performance and also consumes less memory.

Agreed but given the small number of elements I'd guess that the `foreach` loop would take `O(n)` for both.

Reply:
> Are you sure? 

Well no, of course, and it would depend on user code as well.

But if I diagnose the CRUD quickstart, there's only this one an a single use in Narayana. It made me think that it was within reach to avoid it completely in a good amount of cases :)

And of course I won't spend an hour to fix Narayana for this purpose alone, but I'm making a list of optimisations for Narayana and will then send a PR fixing them all.

Reply:
`ArrayList` is almost always better than a `LinkedList`. If you look at memory footprint, forward/backward iteration, random access, and even modifications. Modifying a linked list (especially prepending or inserting into the middle) is theoretically faster than an array in certain computation models (random-access memory, where pointer chasing is not an issue), but in the real world (with memory hierarchy and highly optimized memory copying), arrays win there as well. There may be some cases where a linked list is better in practice, but I believe it's safe to always start with an `ArrayList`.

Reply:
`LinkedList` doesn't make a lot of sense to use all you plan to do is append entries and later on iterate through the results. You are paying the cost of creating new nodes and then traversing them for no benefit whatsoever.

Reply:
Ok, guys I get your point! ;-). Nobody questions that `ArrayList` is the best "default choice". My point is that changes like these look like nitpicking and have no real impact on the performance/memory consumption in this particular use case. Well, we could save few hundreds of bytes in some special cases... Is worth the effort? That said, I'm going to incorporate these `LinkedList` changes in the "cumulative" PR...